rust/arrow/src/buffer.rs

```diff
@@ -140,6 +145,120 @@ impl<T: AsRef<[u8]>> From<T> for Buffer {
     }
 }
 
+impl<'a, 'b> BitAnd<&'b Buffer> for &'a Buffer {
+    type Output = Buffer;
+
+    fn bitand(self, rhs: &'b Buffer) -> Buffer {
+        assert_eq!(
+            self.len(),
+            rhs.len(),
+            "Buffers must be the same size to apply Bitwise AND."
+        );
+
+        // SIMD implementation if available
+        #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+        {
+            let mut result =
+                MutableBuffer::new(self.len()).with_bitset(self.len(), false);
+            let lanes = u8x64::lanes();
```

Comment:
I should have made this generic over the op to clean this up :(,  I'll follow up when I get a chance...